inference-engine/src/mkldnn_plugin/nodes/mkldnn_roll_node.cpp

```diff
@@ -0,0 +1,214 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <string>
+#include <vector>
+#include <cmath>
+#include <mkldnn_extension_utils.h>
+
+#include "mkldnn_roll_node.h"
+#include "ie_parallel.hpp"
+#include "ie_precision.hpp"
+#include "mkldnn/ie_mkldnn.h"
+#include "utils/general_utils.h"
+#include "common/cpu_memcpy.h"
+
+using namespace mkldnn;
+using namespace MKLDNNPlugin;
+using namespace InferenceEngine;
+
+MKLDNNRollNode::MKLDNNRollNode(const InferenceEngine::CNNLayerPtr& layer, const mkldnn::engine& eng, MKLDNNWeightsSharing::Ptr &cache) :
+                MKLDNNNode(layer, eng, cache) {
+    const std::string layerErrorPrefix = "Roll layer with name '" + layer->name + "'";
+    if (layer->insData.size() != numberOfInputs) {
+        IE_THROW() << layerErrorPrefix << " has incorrect number of input/output edges!";
+    }
+
+    /* Data */
+    auto data = layer->insData[DATA_INDEX].lock();
+    if (data == nullptr) {
+        IE_THROW() << layerErrorPrefix << " has nullable data";
+    }
+
+    const auto &dataTensor = data->getTensorDesc();
+    const auto &dataShape = dataTensor.getDims();
+    const auto &dataPrecision = dataTensor.getPrecision();
+
+    if (!MKLDNNPlugin::one_of(dataPrecision, Precision::I8, Precision::U8, Precision::I16, Precision::I32, Precision::FP32, Precision::I64, Precision::BF16)) {
```

Comment:
I think this [solution](https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/mkldnn_plugin/nodes/extract_image_patches.cpp#L294) looks better than specifying all possible precisions.